### PR TITLE
fix(web): place fixed sections on toggle-on and back-fill legacy layouts

### LINKED ISSUE #770: fix(web): toggling on a fixed section absent from a non-empty layout is a silent no-op
## Summary

Turning a fixed section's visibility back on does nothing — permanently and silently — when that section's id is missing from a non-empty `metadata.layout`. The toggle flips `visible: true`, but nothing ever places the id in the layout, and both the document sheet and the PDF render only placed ids. Surfaced by the #731 sections panel (PR #768), which is the first UI that exposes visibility toggles for unplaced sections; the bug itself predates it.

## Problem

What happens vs. what should happen:

- `toggleSectionVisibility` (`apps/web/src/stores/resume.ts`) flips the section's `visible` flag but never inserts the id into `metadata.layout`.
- `layoutPages`/`editorPages` (`apps/web/src/lib/docLayout.ts`) draw only placed ids, so the sheet never shows the section.
- `normalizeResumeForStore` (`apps/web/src/stores/resume.ts:253`) back-fills missing **custom** section ids into a non-empty layout on load, but never fixed ones, so a reload does not repair the state.

Reproduce:

1. Load a legacy resume whose stored layout predates a fixed section (e.g. a layout saved before `coverLetter` existed — `ensureCoverLetterSection` adds the section data but does not place it).
2. Open the document editor's sections panel and toggle the unplaced section on.
3. The switch flips, but the sheet and the rendered PDF never change; toggling off/on and reloading make no difference.

New resumes are immune (the Rust `default_layout` places every fixed section) and template switches self-heal (`layoutForTemplate` carries every id), so the reach is legacy/pruned layouts only.

## Proposed Solution

Either (or both):

- **Toggle-side:** when `toggleSectionVisibility` turns a section on and its id is absent from a non-empty layout, append the id to the layout's first main column in the same store write (one action, one undo entry).
- **Normalization-side:** extend `normalizeResumeForStore` to back-fill missing fixed ids into non-empty layouts the way it already does for custom ids.

The normalization-side fix repairs existing documents on load; the toggle-side fix keeps the invariant that a visible section is always placed. Add a regression test with a resume whose non-empty layout omits a fixed section: toggling on must make it appear in `editorPages()` and stay after normalization round-trips.

## References

- Found during the independent review of PR #768 (issue #731, epic #721).
- Related code: `apps/web/src/stores/resume.ts` (`toggleSectionVisibility`, `normalizeResumeForStore`), `apps/web/src/lib/docLayout.ts` (`layoutPages`, `editorPages`).

### PR DESCRIPTION
**Pull request by TurboCoder13**
## Commit Summary (Conventional Commits)

- Title: `fix(web): place fixed sections on toggle-on and back-fill legacy layouts`

- Type:
  - [ ] feat (minor)
  - [x] fix / perf (patch)
  - [ ] docs
  - [ ] refactor
  - [ ] test
  - [ ] chore / ci / style

- Breaking change:
  - [ ] `!` in title or `BREAKING CHANGE:` footer included

## What's Changing

Toggling a fixed section visible when its id is absent from a non-empty `metadata.layout` was a permanent, silent no-op: the visibility flag flipped, but the document sheet and the PDF draw only placed ids, so nothing ever changed and a reload did not repair it. Surfaced by the #731 sections panel (PR #768); the bug predates it and reaches legacy/pruned layouts (e.g. layouts saved before `coverLetter` existed).

Two-sided fix in the store/normalization layer:

- **Normalization back-fill** — `normalizeResumeForStore` now back-fills missing fixed ids into page 0's first (main) column of a non-empty layout, mirroring the existing custom-id back-fill, so existing documents are repaired on load. The empty-page-0 rebuild branch back-fills absent fixed ids too.
- **Toggle-side placement** — `toggleSectionVisibility` places an unplaced fixed id in the same `setStore(produce(...))` write as the visibility flip: one action, one `markDirty()`, one undo entry. When pages exist but page 0 has no columns, the main column is seeded.
- **`createNewResume` normalizes the factory default** — the local (WASM-unavailable) fallback default places only a subset of fixed sections; every other path into the store (load, import, undo) already normalizes, so this keeps the "a visible section is always placed" invariant on every path.

Back-filled sections stay hidden, so nothing new renders to PDF; `editorPages` draws item sections only when they hold items, so empty back-filled sections do not clutter the sheet either.

## Checklist

- [x] Title follows Conventional Commits
- [x] Tests added/updated
- [ ] Docs updated if user-facing
- [x] Local CI passed (`bun run --cwd apps/web test -- --run`, `bunx tsc --noEmit -p apps/web`, `uv run lintro chk` 100/100)

## Closes

- Closes #770
- Relates to #731
- Relates to #721

## Details

- Regression suite in `apps/web/src/stores/__tests__/resume.test.ts` (`unplaced fixed sections (#770)`): toggle-on places the id in the first main column and it appears in `editorPages()`; toggle-off and empty layouts stay untouched; page-0-without-columns seeds the main column; the flip and placement revert as a single undo entry; the normalization back-fill is idempotent.
- Layout-exactness expectations updated where normalization now back-fills (store sentinel tests, DocEditor sheet/undo tests — the shared fixture's stored layout omits `coverLetter` and `references`, which now surface as hidden chrome instead of being unreachable).
- Full web suite green: 888 tests, 79 files. `tsc` clean, lintro 100/100.
- AI review: CodeRabbit run 1 → 2 major findings (both the empty-page-0 edge), fixed in `ec07485`; run 2 → 0 findings. Greptile unavailable this change set (`free_reviews_limit_reached`).

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved resume layout normalization to restore missing fixed sections, including summaries and cover letters.
  * Ensured newly created resumes receive a valid initial layout.
  * Fixed visibility toggles so restored fixed sections are correctly placed and displayed.
  * Improved handling of empty or incomplete pages and legacy layouts.
* **Tests**
  * Expanded coverage for layout normalization, section visibility, rendering, and undo/redo behavior.

### CODE REVIEW

File: apps/web/src/stores/resume.ts [hunk 1]
```diff
@@ -94,11 +94,14 @@ export function isNotFoundError(error: unknown): boolean {
...
-const FIXED_LAYOUT_SECTION_KEY_SET = new Set<string>([
+/** Every fixed section id a layout can place, in canonical order. */
+const ALL_FIXED_LAYOUT_SECTION_IDS: readonly string[] = [
   "summary",
   "coverLetter",
   ...FIXED_LAYOUT_SECTION_KEYS,
-]);
+];
+
+const FIXED_LAYOUT_SECTION_KEY_SET = new Set<string>(ALL_FIXED_LAYOUT_SECTION_IDS);
```
[the remark this answers is not in the document]
Comment (TurboCoder13) [reply apps/web/src/stores/resume.ts]: Renamed to `ALL_FIXED_LAYOUT_SECTION_ID_SET` (both call sites updated) in 8ea32c3.
